### PR TITLE
[hail] Fix pc_relate bug with non-default call_expr

### PR DESCRIPTION
**Pull request by danking**
resolves https://github.com/hail-is/hail/issues/5263

### CODE REVIEW

File: hail/python/hail/methods/statgen.py [hunk 1]
```diff
@@ -1774,7 +1774,8 @@ def pc_relate(call_expr, min_individual_maf, *, k=None, scores_expr=None,
     mt = matrix_table_source('pc_relate/call_expr', call_expr)
 
     if k and scores_expr is None:
-        _, scores, _ = hwe_normalized_pca(mt.GT, k, compute_loadings=False)
+        mt2 = mt.select_entries(call=call_expr)
+        _, scores, _ = hwe_normalized_pca(mt2.call, k, compute_loadings=False)
```
[the remark this answers is not in the document]
Comment (danking) [reply hail/python/hail/methods/statgen.py]: I was really hoping you didn't say that.

[the remark this answers is not in the document]
Comment (tpoterba) [reply hail/python/hail/methods/statgen.py]: You can make put this in the doctest or something?

**Comment by danking** [comment]
@tpoterba I added a test and an issue that described the problem. I verified the test fails on `dcf43490c` a recent `master` commit.

**Comment by danking** [comment]
@jbloom22 can you give the latest commit on this PR a second set of eyes? I had to fiddle with some tolerances due to https://storage.googleapis.com/hail-ci-0-1/ci/7a0732726e6873e2c0d85fed5183324ac9441d52/194ea22cd9f744a5463340130e799c8a65ca885e/index.html . I expected the test I added (`test_pcrelate_issue_5263`) to be exactly the same but it differed out at the 10th or so position.

**Comment by jbloom22** [comment]
Will do!

**Comment by danking** [comment]
@jbloom22 here

**Comment by jbloom22** [comment]
if you cache the dataset (or set the global seed and generate it, both for the actual and the expected), do you get exactly the same result in your new test?

**Pull request merged**